src/index.ts

```diff
@@ -9,80 +10,117 @@ require('dotenv').config();
+
 
-  const repoMap = await getRepoIds();
+  console.log('');
+  const j1Client = await getClient(clientInput);
+  const repoMap = await getRepoIds(pathToRepos, clientInput);
+  console.log('');
 
-  for (const repoName of Array.from(repoMap.values())) {
+  for (const repoName of Array.from(repoMap.keys())) {
     console.log('Repo: ' + repoName);
 
     const mainRepo = await j1Client.queryV1(
       `FIND CodeRepo WITH name='${repoName}'`,
     );
-    const depsList = getDependencies(repoName, packageScope);
+    const depsList = getDependencies(repoName, packageScope, repoMap);
+    const deployDepsList = getDependenciesYaml(repoName, repoMap);
     if (depsList === undefined) {
-      continue;
-    }
-    if (depsList.length > 0) {
-      for (const dep of depsList) {
-        const fullName = dep.substring(1, dep.indexOf(':'));
-        let depRepo = await j1Client.queryV1(
-          `FIND CodeRepo WITH full_name='${fullName}'`,
-        );
-
-        if (depRepo.length > 0) {
-          const relationshipKey = mainRepo[0].entity._key + '|uses|' + depRepo[0].entity._key;
-          const relationship = await j1Client.createRelationship(
-            relationshipKey,
-            'bitbucket_repo_uses_bitbucket_repo',
-            'USES',
-            mainRepo[0].entity._id,
-            depRepo[0].entity._id,
+      if (deployDepsList === undefined) {
+        continue;
+      }
+    } else if (depsList.length > 0) {
+        for (const dep of depsList) {
+          const fullName = dep.substring(1, dep.indexOf(':'));
+          let depRepo = await j1Client.queryV1(
+            `FIND CodeRepo WITH full_name='${fullName}'`,
           );
-          console.log('Successfully created relationship (' + repoName + ' USES ' + dep + ').');
-          success++;
-        } else {
-          console.log(
-            'Failed to create relationship with ' + dep + 
-            ' (was not found on the graph). Skipped.'
+  
+          if (depRepo.length > 0) {
+            const relationshipKey = mainRepo[0].entity._key + '|uses|' + depRepo[0].entity._key;
+            const relationship = await j1Client.createRelationship(
+              relationshipKey,
+              'bitbucket_repo_uses_bitbucket_repo',
+              'USES',
+              mainRepo[0].entity._id,
+              depRepo[0].entity._id,
+            );
+            console.log('Successfully created relationship (' + repoName + ' USES ' + dep + ').');
+            success++;
+          } else {
+            console.log(
+              'Failed to create relationship with ' + dep + 
+              ' (was not found on the graph). Skipped.'
+            );
+            failure++;
+            if (missingDeps.indexOf(dep) === -1) {
+              missingDeps.push(dep + ' (' + repoName + ').');
+            }
+          }
+        }
+      } else {
+        console.log('Repo has no dependencies of requested scopes');
+      }
+    if (deployDepsList != undefined) {
```

Comment:
Would be nice to break this file up into well-name functions, this is getting hard to follow.

Reply:
😅 Sorry about that. I'll work on that